### PR TITLE
Add test for basic template

### PR DESCRIPTION
**Pull request by jchavarri**
It would be beneficial to test the [basic template](https://github.com/melange-re/melange-basic-template) with latest changes.

This PR adds a step in the esy workflow (only on Ubuntu) to checkout the latest template and build it against the most recent version of melange.

Some notes:
- Pinning `dune` to latest due to Melange requiring dune 3.5 which is not released (should this change be added to the template?)
- Pinning the template to [a branch](https://github.com/melange-re/melange-basic-template/pull/20) due to the above + missing `directory-targets` extension in `dune-project`
- Showcase issue with Stdlib (see comment [below](https://github.com/melange-re/melange/pull/365#issuecomment-1283980006))

### CODE REVIEW

**Comment by jchavarri** [comment]
@anmonteiro I managed to repro the issue with stdlib (finally!), see https://github.com/melange-re/melange/actions/runs/3281718057/jobs/5404123748#step:12:150. 

Build does not fail though, due to #366.

**Comment by anmonteiro** [comment]
Got this to fail with the fix in #367

File: .github/workflows/esy-build.yml [hunk 1]
```diff
@@ -35,3 +35,42 @@ jobs:
...
+      - name: Clone basic template
+        if: ${{ matrix.system == 'ubuntu' }}
+        run: |
+          cd ocaml-tree && npm install && cd ..
+          git clone -b directory-targets https://github.com/jchavarri/melange-basic-template.git
+
+      - name: Remove esy.lock in template
+        if: ${{ matrix.system == 'ubuntu' }}
+        run: |
+          cd ocaml-tree
```
Comment (anmonteiro) [inline .github/workflows/esy-build.yml:48]: this should probably have been `melange-basic-template` @jchavarri

Comment (jchavarri) [reply .github/workflows/esy-build.yml:48]: Right. Fixed in #369.

**Pull request merged**